compiler/noirc_frontend/src/hir/resolution/errors.rs

```diff
@@ -182,6 +182,8 @@ pub enum ResolverError {
     LoopNotYetSupported { span: Span },
     #[error("Expected a trait but found {found}")]
     ExpectedTrait { found: String, span: Span },
+    #[error("Invalid syntax in match pattern")]
+    InvalidSyntaxInPattern { span: Span },
```

Comment:
Would be nice to have a unit test for this, but as this is experimental work I won't block